Off-Top-App/off-top-flutter
# OT-100 | Dynamically pass streaming data to the front-end

**What I Did:**
- Able to pass dynamic topics, userID, audioData into our data pipeline starting from our front-end websocket
- Populating focus score being streamed from our backend
- Updating focus score

**How To Test:**
**PREFACE:** You will need an iOS emulator to test this branch.
- You will need to login with a created email. Which means you will need to fill our MySQL database with our created data:
    1. Go to the google doc and copy the query found (https://docs.google.com/document/d/1yDnXPHTTi3lG7JpxipsTzAXDrPxpBJcawUWIGjWV8sM/edit) into our offTop MySQL database.
    2. Have the spring-boot server, Kafka, zookeeper, and the python producer running 
    3. after you can login to the application **specifically** using `jane@gmail.com`! You may need to click login twice. I'll fix that later. 
    4. You will see the data updating in the spring-boot server when clicking on the recording icon and selecting a topic. Unfortunately, the front-end only works for IOS as we still need to find a way to handle exporting audio files for android. 
- The topic you chose should appear in the spring-boot server like in the photo below
<img width="1680" alt="Screen Shot 2020-04-25 at 4 01 46 AM" src="https://user-images.githubusercontent.com/26909882/80278234-7cd2b480-86a9-11ea-8538-4a1e83aaf342.png">
